apps/browser/lib/api/GoalApi.ts

```diff
@@ -25,4 +25,20 @@ export class GoalApi {
       return goalAdded.id;
     }
   }
+
+  public static async generateTitle(
+    chatId: string,
+    prompt: string
+  ): Promise<string | undefined> {
+    const generateTitle = await fetch(
+      `/api/goal/generate-title?chatId=${chatId}&prompt=${prompt}`
```

Comment:
Have we tested this with large prompts? There may be a character limit we reach with URL params.

Reply:
Also if you agree with moving the API to `/chat/get-name` then we'd want to move this function to a `ChatApi` helper.

Reply:
made it a post request :+1: